Implement external example storage in files

lpar opened the pull request:
Adds support for storing example data in a directory next to the XML
file:

    service.xml
    service/file1

The data is referenced via a _filename attribute on the example:

    <example _filename="file1" ...>

Trailing newlines are trimmed from the file when it is loaded.

Commit:
Implement external example storage in files

Adds support for storing example data in a directory next to the XML
file:

    service.xml
    service/file1

The data is referenced via a _filename attribute on the example:

    <example _filename="file1" ...>

hdm commented on lib/recog/fingerprint.rb:
```diff
@@ -248,13 +248,23 @@ def parse_description(xml)
+        contents = ""
+        fn = File.join(filepath, attrs["_filename"])
+        File.open(fn, "rb") do |file|
+          contents = file.read.chomp
```
There may be cases where the trailing new line is relevant for the match, maybe remove the chomp? For example, if its DER/BER encoded, the last \n may be relevant content (0x10)

hdm reviewed (commented):
I think the with chomp removed this should work for all cases I can think of.

Commit:
Don't remove trailing newlines from example data

Commit:
Fix overlapping character ranges in two regexps

These were causing Ruby to dump out warnings saying "character
class has duplicated range". While the fingerprints still worked,
this fixes them so they don't generate the warnings. All tests
still pass.

Commit:
Fix typo in README
